File: app/src/main/java/work/curioustools/dogfinder/doglisting/api/DogsApi.java
package work.curioustools.dogfinder.doglisting.api;


import org.jetbrains.annotations.NotNull;

import retrofit2.Call;
import retrofit2.http.GET;

public interface DogsApi {

    @GET("api/breeds/image/random")
    @NotNull
    Call<DogImageDto> getImage();

    @NotNull public static final String BASE_URL_WITH_SLASH = "https://dog.ceo/";


}

File: app/src/main/java/work/curioustools/dogfinder/doglisting/api/DogsApi.kt
package work.curioustools.dogfinder.doglisting.api

import retrofit2.Call
import retrofit2.http.GET

interface DogsApi {
    @GET("api/breeds/image/random")
    fun getImage(): Call<DogImageDto>

    companion object{
        const val BASE_URL_WITH_SLASH = "https://dog.ceo/"
    }
}